src/contracts/Protocol/ClaimController.sol

```diff
@@ -19,21 +59,24 @@ contract ClaimController is Ownable, IClaimController, OnlyApi {
         claimLimits[questionId] = claimLimit;
     }
 
-    function claim(uint256 questionId) public onlyOwner {
+    function claim(address user, uint256 questionId) public onlyApi {
         if (claims[questionId].length >= claimLimits[questionId]) revert ClaimLimitReached();
+        if (answers[questionId][user].author == user) revert AlreadyClaimed();
 
-        claims[questionId].push(_msgSender());
-        Answer memory _answer = Answer({state: STATE.CLAIMED, author: _msgSender(), answerURL: "", scoringMetaDataURI: "", finalGrade: 0});
-        answers[questionId][_msgSender()] = _answer;
+        claims[questionId].push(user);
+        Answer memory _answer = Answer({state: STATE.CLAIMED, author: user, answerURL: "", scoringMetaDataURI: "", finalGrade: 0});
+        answers[questionId][user] = _answer;
```

Comment:
I feel like we should add some JS Docs or something else to describe why we have these properties and why not sure where it's documented - looks fine but I don't remember discussing these properties

Reply:
I agree, will look harder at this part when working on answering questions as well. For now again mostly just services to get the most basic version working.

Reply:
Same for below can you add a TODO to add JS Docs and some documentation for this so we can iterate off of